cmd/ci-images-mirror/main.go

```diff
@@ -322,6 +341,89 @@ func main() {
 	logrus.Info("Process ended gracefully")
 }
 
+func validateConfig(config *quayiociimagesdistributor.CIImagesMirrorConfig, registryConfig string, ciOperatorConfigPath string, configFilePathInReleaseRepo string) error {
+	diff, err := getConfigDiff(configFilePathInReleaseRepo, config)
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to get config diff")
+	}
+	errs := []error{}
+	for _, target := range diff {
+		source := config.SupplementalCIImages[target]
+		if source.Image != "" {
+			if strings.HasPrefix(source.Image, "docker.io/") {
+				errs = append(errs, fmt.Errorf("source image %s is from docker.io", source.Image))
+			}
+			if !isAccessible(source.Image, registryConfig) {
+				errs = append(errs, fmt.Errorf("source image %s is not accessible", source.Image))
+			}
+		}
+		if isPromoted(target, ciOperatorConfigPath) {
+			errs = append(errs, fmt.Errorf("target %s would overwrite a promoted tag", target))
+		}
+		if isAccessible(targetToQuayImage(target), registryConfig) {
+			errs = append(errs, fmt.Errorf("target %s already exists in quay", target))
+		}
+	}
+	return utilerrors.NewAggregate(errs)
+}
+
+func getConfigDiff(configFilePathInReleaseRepo string, config *quayiociimagesdistributor.CIImagesMirrorConfig) ([]string, error) {
```

Comment:
I would do it this way: Add a new function `quayiociimagesdistributor.LoadConfigFromReleaseRepo()` to do the job where we use the existing GH client API to get the content of the config file. Moreover, I would imagine that we could take some sharing code from `quayiociimagesdistributor.LoadConfig(o.configFile)`.

Reply:
This was a great idea! I didn't know such a thing existed.